iothub_client/tests/common_dt_e2e/iothubclient_common_dt_e2e.c

```diff
@@ -1052,45 +1105,425 @@ void dt_e2e_get_complete_desired_test_svc_fault_ctrl_kill_Tcp(IOTHUB_CLIENT_TRAN
+    time_t begin_operation;
+    time_t now_time;
+
+    DEVICE_DESIRED_DATA* device_desired_data = _device_desired_data_init();
+
+    if (update_state == DEVICE_TWIN_UPDATE_COMPLETE)
+    {
+        _get_twin_async(_device_twin_callback, device_desired_data);
+
+        callback_received = false;
+        begin_operation = time(NULL);
+
+        // Receive IoT Hub response.
+        while (now_time = time(NULL), difftime(now_time, begin_operation) < MAX_CLOUD_TRAVEL_TIME)
+        {
+            if (Lock(device_desired_data->lock) != LOCK_OK)
+            {
+                ASSERT_FAIL("Lock failed");
+            }
+            else
+            {
+                // Check results.
+                if (device_desired_data->received_callback)
+                {
+                    ASSERT_ARE_EQUAL(DEVICE_TWIN_UPDATE_STATE, device_desired_data->update_state, DEVICE_TWIN_UPDATE_COMPLETE);
+                    ASSERT_IS_NOT_NULL(device_desired_data->cb_payload);
+                    ASSERT_IS_TRUE(device_desired_data->cb_payload_size > 0);
+                    callback_received = device_desired_data->received_callback;
+                    Unlock(device_desired_data->lock);
+                    break;
+                }
+                Unlock(device_desired_data->lock);
+            }
+            ThreadAPI_Sleep(1000);
+        }
+        ASSERT_IS_TRUE(callback_received, "Did not receive the GetTwinAsync callback");
+    }
+    else // DEVICE_TWIN_UPDATE_PARTIAL
+    {
+        int64_t parsed_version;
+        IOTHUB_TWIN_REQUEST_OPTIONS twin_request_options;
+
+        // GetTwinDesiredAsync: No if-not-version.
+        // Expected status 200. Expected version > 0. Expected payload size > 0.
+        _get_twin_desired_async(NULL, _device_twin_section_callback, device_desired_data);
+
+        // Receive IoT Hub response.
+        callback_received = false;
+        begin_operation = time(NULL);
+
+        while (now_time = time(NULL), difftime(now_time, begin_operation) < MAX_CLOUD_TRAVEL_TIME)
+        {
+            if (Lock(device_desired_data->lock) != LOCK_OK)
+            {
+                ASSERT_FAIL("Lock failed");
+            }
+            else
+            {
+                // Check results.
+                if (device_desired_data->received_callback)
+                {
+                    ASSERT_ARE_EQUAL(DEVICE_TWIN_UPDATE_STATE, device_desired_data->update_state, DEVICE_TWIN_UPDATE_PARTIAL);
+                    ASSERT_IS_NOT_NULL(device_desired_data->twin_response);
+                    bool is = device_desired_data->twin_response->is_status_set;
+                    ASSERT_IS_TRUE(device_desired_data->twin_response->is_status_set);
+                    ASSERT_ARE_EQUAL(int64_t, 200, device_desired_data->twin_response->status);
+                    // ASSERT_IS_TRUE(device_desired_data->twin_response->is_version_set);
+                    // ASSERT_IS_TRUE(device_desired_data->twin_response->version > 0);
```

Comment:
needed?

Reply:
This will be needed once the Hub implements our ask to include the version in the response topic.  Commented out so it is not forgotten about.